.circleci/config.yml

```diff
@@ -0,0 +1,178 @@
+      - run:
+          name: Install test dependencies
+          command: mix deps.get
+      - run:
+          name: Create test firmware
+          command: mix firmware
+      - run:
+          name: "Create artifact dir"
+          command: mkdir -p /nerves/deploy/test/artifacts
+      - run:
+          name: Create build context
+          command: >
+            mix json.encode
+            /nerves/deploy/test/artifacts/${CIRCLE_SHA1}.json
+            --sha $CIRCLE_SHA1
+            --repo-org $CIRCLE_PROJECT_USERNAME
+            --repo-name $CIRCLE_PROJECT_REPONAME
+            --branch $CIRCLE_BRANCH
+            --tag $CIRCLE_TAG
+            --pr $CIRCLE_PULL_REQUEST
+            --build-url $CIRCLE_BUILD_URL
+      - run:
+          name: "Sign test fw artifact"
+          command: fwup --sign --private-key $NERVES_FW_PRIV_KEY -i /nerves/build/test/_build/test/nerves/images/test.fw -o /nerves/deploy/test/artifacts/${CIRCLE_SHA1}.fw
+      - store_artifacts:
+          path: /nerves/deploy/test/artifacts
+          destination: test
+      - save_cache:
+          key: nerves/deploy/test-{{ .Revision }}
+          paths:
+            - "/nerves/deploy/test"
+
+  deploy_test:
+    <<: *defaults
+    steps:
+      - run:
+          name: Install dependencies
+          command: |
+            apt-get update
+            apt-get -y install awscli
+      - restore_cache:
+          key: nerves/deploy/test-{{ .Revision }}
+      - deploy:
+          name: Publish notification on SNS
+          command: aws sns publish --topic-arn arn:aws:sns:us-east-1:991424406598:nerves-test-firmware --message "$(cat /nerves/deploy/test/artifacts/${CIRCLE_SHA1}.json)"
+
+  deploy_system:
+    <<: *defaults
+    steps:
+      - restore_cache:
+          key: nerves/deploy/system-{{ .Revision }}
+      - run:
+          name: Install dependencies
+          command: |
+            wget https://github.com/tcnksm/ghr/releases/download/v0.5.4/ghr_v0.5.4_linux_amd64.zip
+            unzip ghr_v0.5.4_linux_amd64.zip
+      - run:
+          name: "Create Release Notes"
+          command: grep -Pazo "(?s)(## ${CIRCLE_TAG}.\*?)[^#]+" /nerves/deploy/system/CHANGELOG.md > /nerves/deploy/system/RELEASE_NOTES
+      - run:
+          name: Deploy artifacts to Github
+          command: ./ghr -draft -t $GITHUB_TOKEN -u nerves-project -r $CIRCLE_PROJECT_REPONAME -b "$(cat /nerves/deploy/system/RELEASE_NOTES)" -replace $CIRCLE_TAG /nerves/deploy/system/artifacts
+
+workflows:
+  version: 2
+  build_test_deploy:
+    jobs:
+      - build_system:
+          context: org-global
+      - build_test:
+          context: org-global
+          requires:
+            - build_system
+      - deploy_test:
+          context: org-global
+          requires:
+            - build_test
+      - deploy_system:
+          context: org-global
+          requires:
+            - build_system
+          filters:
+            branches:
+              only: master
```

Comment:
From the Circle Docs, here is an example of multiple filters
```
workflows:
  version: 2
  build-n-deploy:
    jobs:
      - build:
          filters:
            tags:
              only: /.*/
      - deploy:
          requires:
            - test
          filters:
            tags:
              only: /^config-test.*/
            branches:
              ignore: /.*/
```
https://circleci.com/docs/2.0/workflows/#git-tag-job-execution